### PR TITLE
Subgraph v2

### PR DESCRIPTION
**Pull request by cha0sg0d**

### CODE REVIEW

File: contracts/DFTypes.sol [hunk 1]
```diff
@@ -295,4 +295,84 @@ struct RevealProofArgs {
     uint256[2][2] _b;
     uint256[2] _c;
     uint256[9] _input;
-}
\ No newline at end of file
+}
+
+// THIS NEEDS TO BE UPDATED ANY TIME A CONSTANT IS ADDED
+struct AllConstants {
```
Comment (cha0sg0d) [inline contracts/DFTypes.sol:301]: @henryacaron the other option is to just return `arenaConstants, gameConstants, and snarkConstants`. Might be cleaner and then I just have to parse that on the graph side

Comment (0xhank) [reply contracts/DFTypes.sol:301]: oh yeah this is super gnarly and non upgradeable, I think it makes more sense just to return all the constants from different structs and parse them like we already do for planets and players

File: contracts/facets/DFMoveFacet.sol [hunk 2]
```diff
@@ -44,7 +44,7 @@ contract DFMoveFacet is WithStorage, WithArenaStorage {
         uint256 abandoning
     );
 
-    event GameStarted(uint256 timestamp);
+    event GameStarted(address startPlayer);
```
Comment (0xhank) [inline contracts/facets/DFMoveFacet.sol]: porque

**Review by 0xhank — CHANGES_REQUESTED** [review]
I think the current constants getter function is really unintuitive, and we should just make three calls to the existing getters for that information and parse it. Because the config hash calculator function doesn't care, and just needs to take a couple more params to parse.

File: contracts/facets/DFMoveFacet.sol (hunk 2 above)
Comment (cha0sg0d) [reply contracts/facets/DFMoveFacet.sol]: because we can get the timestamp from the event in subgraph (it's part of the blockchain data)

**Pull request merged**